modules/Menuing.py

```diff
@@ -0,0 +1,214 @@
+import os
+from typing import NoReturn
+
+from modules.Config import config_battle, config_cheats
+from modules.Inputs import PressButton, WaitFrames
+from modules.Memory import ReadSymbol, GetGameState, DecodeString, mGBA, GetCursorOptions, ParseTasks, GetTaskFunc, \
+    GetTask
+from modules.Console import console
+from modules.Enums import GameState, TaskFunc
+from modules.MenuParsers import GetPartyMenuCursorPos, ParseStartMenu, ParsePartyMenuInternal, ParseMenu
+from modules.Pokemon import GetParty
+
+
+def CheckForPickup(encounter_total: int) -> NoReturn:
+    """
+    Function that handles pickup farming.
+    """
+    pokemon_with_pickup = 0
+    pokemon_with_pickup_and_item = []
+    party = GetParty()
+    for i in range(len(party)):
+        if party[i]['ability'] == 'Pickup':
+            pokemon_with_pickup += 1
+            if party[i]['item']['name'] != 'None':
+                pokemon_with_pickup_and_item.append(i)
+    if not config_cheats['pickup']:
+        encounter_threshold_met = encounter_total % config_battle['pickup_check_frequency'] == 0
+        if encounter_threshold_met and pokemon_with_pickup > 0:
+            NavigateStartMenu("POKEMON")
+            while not PartyMenuIsOpen():
+                PressButton(['A'])
+    else:
+        encounter_threshold_met = True
+    try:
+        pickup_threshold = config_battle['pickup_threshold']
+    except:
+        pickup_threshold = 1
+
+    if pickup_threshold > pokemon_with_pickup > 0:
+        pickup_threshold = pokemon_with_pickup
+    if len(pokemon_with_pickup_and_item) >= pickup_threshold and encounter_threshold_met:
+        console.print('Pickup threshold is met! Gathering items.')
+        if not PartyMenuIsOpen():
+            NavigateStartMenu("POKEMON")
```

Comment:
This is getting stuck on 10x and unbounded speeds, scrolling up through the start menu infinitely.